### PR TITLE
Adding Instructions on how to use `uv` and `venv` for Swell installation

### PR DESCRIPTION
**Pull request by Dooruk**
I used @ashiklom's instructions and included them in the documentation. I think `py_lmod_installer` might be confusing for some users so hope this is useful!

Also added a few Github links, feel free to suggest more sources.

### CODE REVIEW

**Review by mranst — COMMENTED** [review]
Nice! This is the method I used when installing swell for the first time, I did find it more straightforward as a new user than the other method.

File: docs/platforms/discover/installing_swell_uv_venv.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+```bash
+mod_swell() {
+  module purge
+  KERNEL_VERSION=$(uname -r)
+  if [[ $KERNEL_VERSION =~ "4.12" ]]; then
+    source /discover/nobackup/projects/gmao/advda/swell/jedi_modules/modules-intel
+  elif [[ $KERNEL_VERSION =~ "5.14" ]]; then
+    source /discover/nobackup/projects/gmao/advda/swell/jedi_modules/modules-intel-sles15
+  else
+    echo "Unkonwn platform: $PLATFORM"
```
Comment (ashiklom) [inline docs/platforms/discover/installing_swell_uv_venv.md]: Bug! That's my bad.

```suggestion
    echo "No matching platform for kernel version: $KERNEL_VERSION"
```

**Review by ashiklom — DISMISSED** [review]
One minor fix (which I think is my bad for giving bad instructions), but otherwise, looks good -- thanks for submitting this!

**Comment by Dooruk** [comment]
This still needs an approval. I can't merge without approval 🙅

**Review by asewnath — APPROVED** [review]

**Comment by rtodling** [comment]
@ashiklom I'd like a chance to try this out before this gets in.  I don't understand why something that seemed to have worked in the past no long seems to work - I insulated a version yesterday - the traditional way of installing - following the direction in SWELL doc. Then, I close my windows, the next day when I came and reloaded the modules - nothing worked - swell is not in the path. 

I will try th uv way to see if any better.

**Comment by ashiklom** [comment]
@rtodling Let me know if you run into trouble! There's a lot of ways these different things can go wrong, depending on exactly what's in your `.bashrc`, etc. If it doesn't work, I could hop on a Teams call to try to debug interactively.

**Pull request merged**